Source/ui_qt/DebugSupport/QtDebugger.cpp

```diff
@@ -0,0 +1,939 @@
+}
+
+std::vector<uint32> QtDebugger::FindCallers(CMIPS* context, uint32 address)
+{
+	std::vector<uint32> callers;
+	for(uint32 i = 0; i < FIND_MAX_ADDRESS; i += 4)
+	{
+		uint32 opcode = context->m_pMemoryMap->GetInstruction(i);
+		uint32 ea = context->m_pArch->GetInstructionEffectiveAddress(context, i, opcode);
+		if(ea == address)
+		{
+			callers.push_back(i);
+		}
+	}
+	return callers;
+}
+
+std::vector<uint32> QtDebugger::FindWordValueRefs(CMIPS* context, uint32 targetValue, uint32 valueMask)
+{
+	std::vector<uint32> refs;
+	for(uint32 i = 0; i < FIND_MAX_ADDRESS; i += 4)
+	{
+		uint32 valueAtAddress = context->m_pMemoryMap->GetWord(i);
+		if((valueAtAddress & valueMask) == targetValue)
+		{
+			refs.push_back(i);
+		}
+	}
+	return refs;
+}
+
+void QtDebugger::CreateAccelerators()
+{
+	//Framework::Win32::CAcceleratorTableGenerator generator;
+	//generator.Insert(ID_VIEW_FUNCTIONS, 'F', FCONTROL | FVIRTKEY);
+	//generator.Insert(ID_VIEW_THREADS, 'T', FCONTROL | FVIRTKEY);
+	//generator.Insert(ID_VM_STEP, VK_F10, FVIRTKEY);
+	//generator.Insert(ID_VM_RESUME, VK_F5, FVIRTKEY);
+	//generator.Insert(ID_VIEW_CALLSTACK, 'A', FCONTROL | FVIRTKEY);
+	//generator.Insert(ID_VIEW_EEVIEW, '1', FALT | FVIRTKEY);
+	//generator.Insert(ID_VIEW_VU0VIEW, '2', FALT | FVIRTKEY);
+	//generator.Insert(ID_VIEW_VU1VIEW, '3', FALT | FVIRTKEY);
+	//generator.Insert(ID_VIEW_IOPVIEW, '4', FALT | FVIRTKEY);
+	//m_nAccTable = generator.Create();
+}
+
+void QtDebugger::DestroyAccelerators()
+{
+	//DestroyAcceleratorTable(m_nAccTable);
+}
+
+/*long QtDebugger::OnCommand(unsigned short nID, unsigned short nMsg, HWND hFrom)
+{
+	switch(nID)
+	{
+	case ID_VIEW_ELF:
+		m_pELFView->Show(SW_SHOW);
+		m_pELFView->setFocus(Qt::ActiveWindowFocusReason);
+		return FALSE;
+		break;
+	return TRUE;
+}*/
+
+/*
+LRESULT QtDebugger::OnWndProc(unsigned int nMsg, WPARAM wParam, LPARAM lParam)
+{
+	switch(nMsg)
+	{
+	case WM_EXECUNLOAD:
+		OnExecutableUnloadingMsg();
+		return FALSE;
+		break;
+	case WM_EXECCHANGE:
+		OnExecutableChangeMsg();
+		return FALSE;
+		break;
+	case WM_MACHINESTATECHANGE:
+		OnMachineStateChangeMsg();
+		return FALSE;
+		break;
+	case WM_RUNNINGSTATECHANGE:
+		OnRunningStateChangeMsg();
+		return FALSE;
+		break;
+	}
+	return CMDIFrame::OnWndProc(nMsg, wParam, lParam);
+}*/
+
+void QtDebugger::OnFunctionsViewFunctionDblClick(uint32 address)
+{
+	GetDisassemblyWindow()->SetAddress(address);
+}
+
+void QtDebugger::OnFunctionsViewFunctionsStateChange()
+{
+	GetDisassemblyWindow()->HandleMachineStateChange();
+	GetCallStackWindow()->HandleMachineStateChange();
+}
+
+void QtDebugger::OnThreadsViewAddressDblClick(uint32 address)
+{
+	auto disAsm = GetDisassemblyWindow();
+	disAsm->SetCenterAtAddress(address);
+	disAsm->SetSelectedAddress(address);
+}
+
+void QtDebugger::OnExecutableChange()
+{
+	//SendMessage(m_hWnd, WM_EXECCHANGE, 0, 0);
+}
+
+void QtDebugger::OnExecutableUnloading()
+{
+	//SendMessage(m_hWnd, WM_EXECUNLOAD, 0, 0);
```

Comment:
do `SendMessage()` sole function is to trigger `OnWndProc()`? so we can just map these calls directly to their equivalent call?